### ISSUE TITLE
MultiSelect issues when used in a customized Grid filter menu

### ISSUE THREAD

**Issue by dtopalov** [telerik/kendo-angular#1068]
There are several issues with the MultiSelect and its focus when the component is used within a customized Grid Filter menu:

[http://plnkr.co/edit/ss8rpAgmC8uZB3zfYlP7?p=preview](http://plnkr.co/edit/ss8rpAgmC8uZB3zfYlP7?p=preview)

1) Keyboard navigation is not working when the MultiSelect is opened, the user cannot navigate through the list of options with arrow up/down keys

2) If no value is selected and the list is scrolled, values cannot be selected and the list is instead closed (without firing the **valueChange** event).

If a value is selected before the list is scrolled, then values can be selected as expected after scrolling as well.

**Comment by glapointe** [comment]
I'm having the same issues. The issue with not being able to select an item once you scroll is a show stopper for me. The plunker in the original isn't working at all for me any more but here's one I just created to replicate #2 (simply filter by Category and scroll the multi-select and you won't be able to pick any items in the list): https://plnkr.co/edit/bWYLvFE8qT9jIM7JFFaS?p=preview

**Comment by ipeshev** [comment]
Fixed in v2.1.0-dev.201802151408

**Comment by tsvetomir** [comment]
Released in v2.1.0

**Comment by brybry44** [comment]
I am having this exact issue still. I only recently installed Kendo 2 months ago and my dropdowns package version is at 3.1. In my apps package.json under dependencies, dropdowns package was listed with version "2-3" Does that need to be changed or is this still an issue?

**Comment by brybry44** [comment]
I ran an npm update --all and it updated all of my dependencies to the latest versions, but the problem still persists.

**Comment by brybry44** [comment]
Hey @daniloacim if you are still having this issue I have found a fix.

The issue is with the valueFilters returning an empty array when you are trying to click on a value. 
If you check the length of the array before hand and return null if empty the problem is fixed.

` public stateFilters(filter: CompositeFilterDescriptor): FilterDescriptor[] {
        var filterArray = flatten(filter).map(({ value }) => value)
        if(filterArray.length === 0 )
        {
            return null
        }
        else {
           return flatten(filter).map(({ value }) => value)
        }
    }`

**Comment by valchev** [comment]
Hi guys,

I can confirm that there is a regression and the bug is present in the recent versions of the package.
We will investigate the issue further and provide a fix as soon as possible.

**Comment by dtopalov** [comment]
The issue is caused by the fact that the MultiSelect value is replaced with another array instance upon each interaction with the component (keyboard navigation, item selection, etc.)

The Grid filtering demo was updated so that the MultiSelect value is always the same array instance (that is mutated accordingly):

[https://stackblitz.com/edit/angular-klu4s6-ncfcrf?file=app/app.component.ts](https://stackblitz.com/edit/angular-klu4s6-ncfcrf?file=app/app.component.ts)

**Comment by dimitar-pechev** [comment]
Closing this, as we don't intend to change the current **MultiSelect** beahvior - handling value change on each change detection cycle is not optimal.

Either use the provided example by @dtopalov (also present in [the docs](https://www.telerik.com/kendo-angular-ui/components/grid/filtering/built-in-template/#toc-customizing-filter-menus)) to use a single array instance, or create a custom component/directive and update the **MultiSelect** data only `onOnit`. Here's [an example](https://stackblitz.com/edit/angular-bhav1r) with a directive.
